internal/resources/schemas/lb_profile_schemas.go

```diff
@@ -0,0 +1,426 @@
+					Required:         true,
+					Description:      "Network Loadbalancer Supported values are `LBHttpProfile`,`LBFastTcpProfile`, `LBFastUdpProfile`, `LBClientSslProfile`,`LBServerSslProfile`, `LBCookiePersistenceProfile`,`LBGenericPersistenceProfile`,`LBSourceIpPersistenceProfile`",
+				},
+				"share_persistence": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "ntlm_authentication for Network Load balancer Profile",
+				},
+				"ha_persistence_mirroring": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "ha_persistence_mirroring for Network Load balancer Profile",
+				},
+				"persistence_entry_timeout": {
+					Type:             schema.TypeInt,
+					Optional:         true,
+					Default:          300,
+					ValidateDiagFunc: validations.IntBetween(1, 2147483647),
+					Description:      "persistence_entry_timeout for Network Load balancer Profile",
+				},
+				"purge_entries_when_full": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     true,
+					Description: "purge_entries_when_full for Network Load balancer Profile",
+				},
+			},
+		},
+	}
+}
+
+func GenericProfileSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Generic Profile configuration",
+		MaxItems:    1,
+		ExactlyOneOf: []string{"http_profile", "tcp_profile", "udp_profile",
+			"cookie_profile", "sourceip_profile", "generic_profile",
+			"client_profile", "server_profile"},
+		ConflictsWith: []string{"http_profile", "tcp_profile",
+			"udp_profile", "cookie_profile", "sourceip_profile",
+			"client_profile", "server_profile"},
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"service_type": {
+					Type:             schema.TypeString,
+					ValidateDiagFunc: validations.StringInSlice([]string{"LBHttpProfile", "LBFastTcpProfile", "LBFastUdpProfile", "LBClientSslProfile", "LBServerSslProfile", "LBCookiePersistenceProfile", "LBGenericPersistenceProfile", "LBSourceIpPersistenceProfile"}, false),
+					Required:         true,
+					Description:      "Network Loadbalancer Supported values are `LBHttpProfile`,`LBFastTcpProfile`, `LBFastUdpProfile`, `LBClientSslProfile`,`LBServerSslProfile`, `LBCookiePersistenceProfile`,`LBGenericPersistenceProfile`,`LBSourceIpPersistenceProfile`",
+				},
+				"share_persistence": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "ntlm_authentication for Network Load balancer Profile",
+				},
+				"ha_persistence_mirroring": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "ha_persistence_mirroring for Network Load balancer Profile",
+				},
+				"persistence_entry_timeout": {
+					Type:     schema.TypeInt,
+					Optional: true,
+					//Default:          300,
```

Comment:
i see that u have removed default field for some of the schema fields. is it intentional ? and why ?

Reply:
we need these default for particular profile types